src/aws/flb_aws_util.c

```diff
@@ -927,3 +932,73 @@ flb_sds_t flb_get_s3_key(const char *format, time_t time, const char *tag,
         }
         return NULL;
 }
+
+/*
+ * Adjust the time_key output format according to time_key_format.
+ */
```

Comment:
Please update description to clearly define function contract. I think this function has no concept of keys. Just time and strings.